annubes/task.py

```diff
@@ -165,42 +162,32 @@ def generate_trials(
         trials["outputs"] = self._outputs
         return trials
 
-    def _rescale(
+    def _minmaxscaler(
         self,
-        input_: float,
-        coeff: float,
-        min_intensity: float | None = None,
-        max_intensity: float | None = None,
+        input_: NDArray[np.float32],
+        input_range: tuple[float, float] = (0, 1),
     ) -> float:
-        """Rescale `input_` value along (0,`coeff`) if `coeff` is non-zero.
+        """Rescale `input_` array to a given range.
 
         Rescaling happens as follows:
-            coeff * (input_ - min_intensity) / (max_intensity - min_intensity)
+
+            `X_std = (input_ - input_.min()) / (input_.max() - input_.min())`
+            `X_scaled = X_std * (max - min) + min`
+            where min, max = range.
+        The logic is the same as that of `sklearn.preprocessing.MinMaxScaler` estimator. Each array is rescaled to the
```

Comment:
Isn't this function the same as sklearn.preprocessing.minmax_scale()`? Why not just import that [function](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.minmax_scale.html#sklearn.preprocessing.minmax_scale) rather than defining our own version of it?

Reply:
Because the axis used by sklearn are different (axis=0); here we have an array with shape (number of trials, number of data points in one time series, number of time series). This is not the shape that MinMaxScaler expects, which has been developed for standard tabular data (e.g., shapes like n_samples, n_features). In the beginning, this was my idea as well (i.e., import directly scikit-learn functions), but I couldn't find a straightforward way to actually do it. Feel free to try yourself of course!! It would be definitely better to just reuse an already-existing function

Reply:
I understand.